src/Open3D/Visualization/Rendering/Filament/FilamentResourceManager.cpp

```diff
@@ -0,0 +1,718 @@
+                                            descriptor.baseColor.g,
+                                            descriptor.baseColor.b};
+    materialInstance->setParameter("baseColor", filament::RgbType::sRGB,
+                                   baseColor);
+
+#define TRY_ASSIGN_MAP(map)                                                    \
+    {                                                                          \
+        if (descriptor.map && descriptor.map->HasData()) {                     \
+            auto hMapTex = CreateTexture(descriptor.map);                      \
+            if (hMapTex) {                                                     \
+                materialInstance->setParameter(#map, textures_[hMapTex].get(), \
+                                               sampler);                       \
+                dependencies_[handle].insert(hMapTex);                         \
+            }                                                                  \
+        }                                                                      \
+    }
+
+    materialInstance->setParameter("baseRoughness", descriptor.baseRoughness);
+    materialInstance->setParameter("baseMetallic", descriptor.baseMetallic);
+
+    TRY_ASSIGN_MAP(albedo);
+    TRY_ASSIGN_MAP(normalMap);
+    TRY_ASSIGN_MAP(ambientOcclusion);
+    TRY_ASSIGN_MAP(metallic);
+    TRY_ASSIGN_MAP(roughness);
+
+#undef TRY_ASSIGN_MAP
+
+    return handle;
+}
+
+TextureHandle FilamentResourceManager::CreateTexture(const char* path) {
+    std::shared_ptr<geometry::Image> img;
+
+    if (path) {
+        img = io::CreateImageFromFile(path);
+    } else {
+        utility::LogWarning("Empty path for texture loading provided");
+    }
+
+    return CreateTexture(img);
+}
+
+TextureHandle FilamentResourceManager::CreateTexture(
+        const std::shared_ptr<geometry::Image>& img) {
+    TextureHandle handle;
+    if (img->HasData()) {
+        auto texture = LoadTextureFromImage(img);
+
+        handle = RegisterResource<TextureHandle>(engine_, texture, textures_);
+    }
+
+    return handle;
+}
+
+TextureHandle FilamentResourceManager::CreateTexture(
+        const geometry::Image& image) {
+    TextureHandle handle;
+    if (image.HasData()) {
+        auto copy = std::make_shared<geometry::Image>(image);
+
+        auto texture = LoadTextureFromImage(copy);
+
+        handle = RegisterResource<TextureHandle>(engine_, texture, textures_);
+    }
+
+    return handle;
+}
+
+TextureHandle FilamentResourceManager::CreateTextureFilled(
+        const Eigen::Vector3f& color, size_t dimension) {
+    TextureHandle handle;
+    auto texture = LoadFilledTexture(color, dimension);
+    handle = RegisterResource<TextureHandle>(engine_, texture, textures_);
+
+    return handle;
+}
+
+IndirectLightHandle FilamentResourceManager::CreateIndirectLight(
+        const ResourceLoadRequest& request) {
+    IndirectLightHandle handle;
+
+    if (false == request.path.empty()) {
```

Comment:
if (!request.path.empty()) ?

Reply:
I will defer to the original developer's sense of style, even though it isn't idomatic C++. (Unless our style guide forbids it?)